### PR TITLE
Feat: expires time

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -270,6 +270,10 @@ func CreateAuthorizationRequestWithMessage(reason, message, sender,
 		Scope:       []protocol.ZeroKnowledgeProofRequest{},
 	}
 	request.From = sender
+	createTime := time.Now().Unix()
+	expiresTime := time.Now().Add(time.Hour).Unix()
```
Comment (demonsh) [inline auth.go:274]: Shouldn't expire time be configurable?

Comment (volodymyr-basiuk) [reply auth.go]: Have added options.

File: auth.go [hunk 2]
```diff
@@ -559,7 +592,14 @@ func (v *Verifier) FullVerify(
...
 ) (*protocol.AuthorizationResponseMessage, error) {
-
+	cfg := pubsignals.VerifyConfig{}
+	for _, o := range opts {
+		o(&cfg)
+	}
+	if (cfg.AllowExpiredMessages == nil || !*cfg.AllowExpiredMessages) &&
+		request.ExpiresTime != nil && time.Now().After(time.Unix(*request.ExpiresTime, 0)) {
+		return nil, errors.New("Authorization request message is expired")
+	}
```
Comment (vmidyllic) [inline auth.go:602]: VerifyAuthResponse is called in this function, don't need to write check twice

File: auth.go [hunk 3]
```diff
@@ -270,7 +289,14 @@ func CreateAuthorizationRequestWithMessage(reason, message, sender,
...
 	request.From = sender
-
+	createTime := time.Now().Unix()
+	request.CreatedTime = &createTime
+	var expiresTime *int64
+	if cfg.ExpiresTime != nil {
+		expiresTimeUnix := cfg.ExpiresTime.Unix()
+		expiresTime = &expiresTimeUnix
+	}
+	request.ExpiresTime = expiresTime
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply auth.go:299]: yes, I guess it's correct behaviour ?

File: auth.go (hunk 2 above)
Comment (volodymyr-basiuk) [reply auth.go:602]: Should we verify response message inside VerifyAuthResponse ?
So in full verify it checks request and response verified inside VerifyAuthResponse.

Comment (vmidyllic) [reply auth.go:602]: ah..ok maybe you are right

File: auth.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (vmidyllic) [reply auth.go:299]: confirmed with @demonsh

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**